### DISCUSSION TITLE
Global, per project, user activity statistics

### DISCUSSION THREAD

**Discussion by silopolis** [WeblateOrg/weblate#8735] [Q&A]
Hi,

After spending some time this month on translations, I wanted to get personal activity statistics for that period. Something like the https://hosted.weblate.org/user/<user>/#activity page, but with the breakdown obtained with the Project/Insights/Translation Reports/Contributor Stats.

After trying to play with the REST API (brand new to me), I resigned to use the form for each project I contributed to.

Is there a way such data could be obtained more easily?

TY
J

**Comment by nijel** [comment]
Not right now, see also https://github.com/WeblateOrg/weblate/issues/4731 and https://github.com/WeblateOrg/weblate/issues/2752

**Comment by silopolis** [comment]
> Not right now, see also #4731
> <https://github.com/WeblateOrg/weblate/issues/4731> and #2752
> <https://github.com/WeblateOrg/weblate/issues/2752>
>

OK. Good to see things have already reached the backlog :)

So, hoping to push the cards a step (or two !) up the ladder, from the
translator PoV, this would be something both very valuable and motivating.
Further, it would bring an edge over the two other most used TMSs by the
Open-Source community.

Coding platforms, specially GH, take great care to offer features allowing
users to demonstrate their work and worth. With this, some users' profiles
are now worthier than the best resume. It would be fantastic if TMSs, all
the more the best Open-Source one ;-) , could offer the same kind of
features.
As an example, I spent a tediously long time yesterday afternoon building a
simple chart to summarize my translation activity (see joint chart), and
it's a despair to think I'll have to do that each month...

Contrary to some comments in the linked issues, AFAIC data seems to be here
as I can get them by generating a report for each project. I'm 'just'
missing them to be regrouped somewhere, somehow, in my user profile, and/or
with some `GET /api/users/(str: username)/statistics/` collection
regrouping the data from the reports of all projects a user is member/has
contributed to would be perfect

Thanks in advance for your consideration
TY

**Reply by nijel** [reply]
To be precise – the data is there, but the aggregation is expensive to be used on the profile directly. That's why we collect metrics for project/component/translation and use that for display. Something similar needs to be implemented for user stats.

**Reply by silopolis** [reply]
Your message made me look at this again and discover in #4318 that `/api/users/(str: username)/statistics/` was already here holding global user stats!
A 'simple' breakdown by project/component/translation would already be great and would allow us to better valuate our contributions :)
Of course, if it'd get more parameters like time frames…

**Reply by nijel** [reply]
Yes, that's the only aggregation we do right now. The more detailed data is not aggregated and thus expensive to display.

**Comment by silopolis** [comment]
>
> The more detailed data is not aggregated and thus expensive to display.
>

API would be nice for starters 😋

**Reply by nijel** [reply]
It's there: https://docs.weblate.org/en/latest/api.html#changes

**Reply by silopolis** [reply]
Indeed!
Could make it with a little massage…

Thank you :)
